src/sas/qtgui/Perspectives/Fitting/FittingWidget.py

```diff
@@ -2940,8 +2940,7 @@ def onMainParamsChange(self, top, bottom):
             # don't try to update multiplicity counters if they aren't there.
             # Note that this will fail for proper bad update where the model
             # doesn't contain multiplicity parameter
-            if self.kernel_module.params.get(parameter_name, None):
```

Comment:
Suspicious: `bool(0.0)` is false, so if the current value of the parameter is 0 then this code wasn't updating it.

I didn't look at the context, but this change may fix other bugs.

Reply:
There are a number of issues when params become 0. We should also test to see if this solves https://github.com/SasView/sasview/issues/3053

Reply:
Out of curiosity, I decided to test and, unless I am mistaking the issue, this should now fix #3053.